compiler/testData/diagnostics/testsWithStdLib/contracts/smartcasts/deeplyNested.fir.kt

```diff
@@ -95,7 +95,7 @@ fun branchedAndNestedWithNativeOperators(x: Any?, y: Any?) {
                     equalsTrue(isInt(y) && isString(y))                          // y is Int, String
             )
             &&
-            (1 == 2 || y is Int || isString(y))
+            (1 == 2 || <!USELESS_IS_CHECK!>y is Int<!> || isString(y))
```

Comment:
Note that this is `&&` clause, which means the previous one is `true`, where both sides of `||` include `y is Int` or `isInt(y)` with contract `y is Int`. Hence, this is what FIR smartcast detects well (while FE 1.0 didn't warn anything).